tests/servo/test_servo.py

```diff
@@ -1,18 +1,68 @@
 import os
+import xml.etree.ElementTree as ET
+from ingenialink.utils._utils import get_drive_identification
+
+
+def _clean(filename):
+    if os.path.isfile(filename):
+        os.remove(filename)
 
 
 def test_save_configuration(connect_to_slave):
     servo, net = connect_to_slave
     assert servo is not None and net is not None
 
     filename = 'temp_config'
+    _clean(filename)
 
     servo.save_configuration(filename)
 
     assert os.path.isfile(filename)
 
-    if os.path.isfile(filename):
-        os.remove(filename)
+    with open(filename, 'r', encoding='utf-8') as xml_file:
+        tree = ET.parse(xml_file)
+    root = tree.getroot()
+
+    device = root.find('Body/Device')
+    prod_code, rev_number = get_drive_identification(servo)
+    if 'ProductCode' in device.attrib and prod_code is not None:
+        assert int(device.attrib.get('ProductCode')) == prod_code
+    if 'RevisionNumber' in device.attrib and rev_number is not None:
+        assert int(device.attrib.get('RevisionNumber')) == rev_number
+
+    assert device.attrib.get("PartNumber") == servo.dictionary.part_number
+    assert device.attrib.get("Interface") == servo.dictionary.interface
+    assert device.attrib.get("firmwareVersion") == servo.dictionary.firmware_version
+    # TODO: check name and family? These are not stored at the dictionary
+
+    saved_registers = root.findall('./Body/Device/Registers/Register')
+    assert len(saved_registers) > 0
+    subnodes = [0, 1]
+
+    for saved_register in saved_registers:
+        subnode = int(saved_register.attrib.get('subnode'))
+        assert int(subnode) >= 0
+        if subnode not in subnodes:
+            continue
+
+        reg_id = saved_register.attrib.get('id')
+        registers = servo.dictionary.registers(subnode=subnode)
+
+        assert reg_id in registers
+
+        storage = saved_register.attrib.get('storage')
+        if storage is not None:
```

Comment:
With Python 3.8+ you could do:

```
reg_storage = registers[reg_id].storage
if (saved_storage := saved_register.attrib.get('storage')) is not None:
    reg_storage = str(reg_storage)
assert reg_storage == saved_storage

Reply:
Interesting, but the library supports py36 and 38 as well, right?

Reply:
Yes, the comment was just to show how we could simplify this part of the code by using the walrus operator.